test/upgrade/2.0.0-2.1.0.js

```diff
@@ -0,0 +1,1108 @@
+        await fundsHandler
+          .connect(sellers[j].wallet)
+          .depositFunds(sellers[j].seller.id, offer.exchangeToken, sellerPool, { value: msgValue });
+      }
+    }
+
+    // group some offers
+    let groupId = 0;
+    for (let i = 0; i < sellers.length; i = i + 2) {
+      const seller = sellers[i];
+      const group = new Group(++groupId, seller.seller.id, seller.offerIds); // group all seller's offers
+      const condition = mockCondition({
+        tokenAddress: mockConditionalToken.address,
+        maxCommits: "10",
+      });
+      await groupHandler.connect(seller.wallet).createGroup(group, condition);
+
+      groups.push(group);
+    }
+
+    // create some twins
+    let twinId = 0;
+    for (let i = 1; i < sellers.length; i = i + 2) {
+      const seller = sellers[i];
+      await mockTwinTokens[0].connect(seller.wallet).setApprovalForAll(protocolDiamondAddress, true);
+      await mockTwinTokens[1].connect(seller.wallet).setApprovalForAll(protocolDiamondAddress, true);
+      // create multiple ranges
+      const twin = mockTwin(rando.address, TokenType.NonFungibleToken);
+      twin.amount = "0";
+      for (let j = 0; j < 7; j++) {
+        twin.tokenId = `${j * 10000 + i * 100}`;
+        twin.supplyAvailable = `${10 * (i + 1)}`;
+        (twin.tokenAddress = mockTwinTokens[j % 2].address), // oscilate between twins
+          (twin.id = ++twinId);
+        await twinHandler.connect(seller.wallet).createTwin(twin);
+
+        twins.push(twin);
+      }
+    }
+
+    // commit to some offers: first buyer commit to 1 offer, second to 2, third to 3 etc
+    await setNextBlockTimestamp(Number(offers[offers.length - 1].offerDates.validFrom)); // When latest offer is valid, also other offers are valid
+    let exchangeId = 0;
+    for (let i = 0; i < buyers.length; i++) {
+      for (let j = i; j < buyers.length; j++) {
+        const offerId = i + j + 1; // some offers will be picked multiple times, some never
+        const offerPrice = offers[offerId - 1].offer.price;
+        const buyerWallet = buyers[j].wallet;
+        let msgValue;
+        if (offers[offerId - 1].offer.exchangeToken == ethers.constants.AddressZero) {
+          msgValue = offerPrice;
+        } else {
+          // approve token transfer
+          msgValue = 0;
+          await mockToken.connect(buyerWallet).approve(protocolDiamondAddress, offerPrice);
+        }
+        await mockToken.mint(buyerWallet.address, offerPrice);
+        await exchangeHandler.connect(buyerWallet).commitToOffer(buyerWallet.address, offerId, { value: msgValue });
+        exchanges.push({ exchangeId: ++exchangeId, offerId: offerId, buyerIndex: j });
+      }
+    }
+
+    // redeem some vouchers #4
+    for (const id of [2, 5, 11, 8]) {
+      const exchange = exchanges[id];
+      await exchangeHandler.connect(buyers[exchange.buyerIndex].wallet).redeemVoucher(exchange.exchangeId);
+    }
```

Comment:
You could use Promise.all to optimize here

```javascript
Promise.all([2, 5, 11, 8].map(id => {
  const exchange = exchanges[id];
  return exchangeHandler.connect(buyers[exchange.buyerIndex].wallet).redeemVoucher(exchange.exchangeId);
}))
```

Reply:
I thought nonce would not be a problem since we are using different accounts in each promise but I guess maybe the problem is with the block nonce and not with the account nonce.